File: cpp-answers/66.加一.cpp
/*
 * @Author: your name
 * @Date: 2020-05-30 01:00:00
 * @LastEditTime: 2020-05-30 01:03:19
 * @LastEditors: your name
 * @Description: In User Settings Edit
 * @FilePath: \leetcode刷题\66.加一.cpp
 */
/*
 * @lc app=leetcode.cn id=66 lang=cpp
 *
 * [66] 加一
 */

// @lc code=start
class Solution
{
public:
    vector<int> plusOne(vector<int> &digits)
    {
        bool carry = true;
        for (int i = digits.size() - 1; i >= 0 && carry; i--)
        {
            carry = (++digits[i] %= 10) == 0;
        }

        if (carry)
        {
            digits.insert(digits.begin(), 1);
        }

        return digits;
    }
};
// @lc code=end

File: rust-answers/66.加一.rs
/*
 * @lc app=leetcode.cn id=66 lang=rust
 *
 * [66] 加一
 */

// @lc code=start
impl Solution {
    pub fn plus_one(mut digits: Vec<i32>) -> Vec<i32> {
        for x in digits.iter_mut().rev() {
            match *x == 9 {
                true => *x = 0,
                false => {
                    *x += 1;
                    return digits;
                }
            }
        }
        digits.insert(0, 1);
        digits
    }
}
// @lc code=end

